Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/features/sysupdate/ui/SystemUpdateProgressBarDialogFragment.java

```diff
@@ -32,32 +34,58 @@ public Dialog onCreateDialog(@Nullable Bundle savedInstanceState)
     SystemUpdateViewModel viewModel =
             new ViewModelProvider(requireActivity()).get(SystemUpdateViewModel.class);
 
-    DialogProgressBinding dialogProgressBinding =
-            DialogProgressBinding.inflate(getLayoutInflater());
+    DialogProgressBinding dialogProgressBinding;
+    DialogProgressTvBinding dialogProgressTvBinding;
 
     // We need to set the message to something here, otherwise the text will not appear when we set it later.
-    AlertDialog progressDialog = new MaterialAlertDialogBuilder(requireContext())
-            .setTitle(getString(R.string.updating))
-            .setMessage("")
-            .setNegativeButton(getString(R.string.cancel), null)
-            .setView(dialogProgressBinding.getRoot())
-            .setCancelable(false)
-            .create();
-
-    viewModel.getProgressData()
-            .observe(this, (@Nullable
-                    Integer progress) -> dialogProgressBinding.updateProgress.setProgress(
-                    progress));
-
-    viewModel.getTotalData().observe(this, (@Nullable Integer total) ->
+    MaterialAlertDialogBuilder progressDialogBuilder =
+            new MaterialAlertDialogBuilder(requireContext())
+                    .setTitle(getString(R.string.updating))
+                    .setMessage("")
+                    .setNegativeButton(getString(R.string.cancel), null)
+                    .setCancelable(false);
+
+    // TODO: Remove dialog_progress_tv if we switch to an AppCompatActivity for leanback
+    if (TvUtil.isLeanback(requireContext()))
```

Comment:
This is technically not correct. You should be checking if the activity is the TvMainActivity (or if the activity inherits from AppCompat), not whether the device supports leanback, since it is possible to launch the "wrong" activity out of the two. (Very unlikely that someone who isn't a developer would launch TvMainActivity on a non-leanback device, though.)